src/cowstr.rs

```diff
@@ -0,0 +1,37 @@
+use std::{borrow::Cow, ops::Deref};
+
+use serde::Deserialize;
+
+/// A wrapper around `Cow<str>` that implements `Deserialize` and can deserialize
+/// string keys into Cow::Borrowed when possible.
+/// 
+/// This is because serde always deserializes strings into `Cow::Owned`.
+/// https://github.com/serde-rs/serde/issues/1852#issuecomment-559517427
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Deserialize)]
```

Comment:
`Ord` and`PartialOrd` would also make sense here

Reply:
added